src/core/ComboBox/ComboBox.tsx

```diff
@@ -103,25 +119,44 @@ export const ComboBox = <T,>(props: ComboBoxProps<T>) => {
     [options, id],
   );
 
+  const userOnChange = React.useRef(onChange);
+
   const memoizedItems = React.useMemo(
     () =>
-      options.map(({ label, value, ...rest }, index) => (
-        <MenuItem
-          id={getOptionId(index)}
-          key={getOptionId(index)}
-          value={value}
-          role='option'
-          onClick={(value: T) => {
+      options.map((option, index) => {
+        const { label, value, ...rest } = option;
+        const additionalProps = {
+          value: value,
```

Comment:
Is `value` a valid prop/attribute to pass to user rendered element?

Reply:
I think it might be ok. If not used it will just appear in the DOM 🤷‍♀️